### PR TITLE
testcase/kernel: fix kernel tc roundrobin fail on ESP32

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
@sunghan-chang

### CODE REVIEW

**Comment by jeongchanKim** [comment]
Please split the commit for `os/arch/xtensa/src/esp32/esp32_timerisr.c`.
Because this file is not for tc.

**Comment by seinfra** [comment]
Target : [ae9351bd4c8464554b06219ea1dc0972770a7cb2] - Code Rule Check Result:
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxO)
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxW)
os/arch/xtensa/src/esp32/esp32_timerisr.c:128: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/esp32_timerisr.c:129: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [ae9351bd4c8464554b06219ea1dc0972770a7cb2] - Code Rule Check (C++) OK.

File: os/arch/xtensa/src/esp32/esp32_timerisr.c [hunk 1]
```diff
@@ -169,7 +169,7 @@ static int esp32_timerisr(int irq, uint32_t *regs, FAR void *arg)
 		 */
 
 		diff = xtensa_getcount() - compare;
-	} while (diff < divisor);
+	} while (diff > divisor);
```
Comment (SRCN-LiQin) [inline os/arch/xtensa/src/esp32/esp32_timerisr.c]: how about diff==divisor case? it seems need to loop one more time

Comment (ChinaTizenRT) [reply os/arch/xtensa/src/esp32/esp32_timerisr.c]: ok, it is corner case, should be consider.tks

**Comment by seinfra** [comment]
Target : [faa7225a8126f7dcc68fcfdc9f7ed5358962c9bf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [faa7225a8126f7dcc68fcfdc9f7ed5358962c9bf] - Code Rule Check Result:
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxO)
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxW)
os/arch/xtensa/src/esp32/esp32_timerisr.c:128: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/esp32_timerisr.c:129: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by sunghan-chang** [comment]
@ChinaTizenRT 
1. Please fill commit description for all of commits. That's our commit rules. Please find our rules on our wiki.
https://github.com/Samsung/TizenRT/wiki/Documentations#commit-rules
2. 1st commit has two different kinds changes. Please split it as change a stack size of roundrobin testcase and enable roundrobin testcase on testcase defconfig of esp32.

**Comment by sunghan-chang** [comment]
And could you fix coding rules violations as shown above with new commit?

**Comment by seinfra** [comment]
Target : [3b8a971940926c6f471ed188c2ff396e3d45e56b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3b8a971940926c6f471ed188c2ff396e3d45e56b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [12fb277a5a633bfe2b6a595d9663d9147cb6649f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [12fb277a5a633bfe2b6a595d9663d9147cb6649f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [074a4d6157b3097ab97bde604c522b5013b00f95] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [074a4d6157b3097ab97bde604c522b5013b00f95] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3be9d2a4f1cde14fc2d5c0c0b6d074759ca58010] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3be9d2a4f1cde14fc2d5c0c0b6d074759ca58010] - Code Rule Check Result:
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxO)
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxW)
os/arch/xtensa/src/esp32/esp32_timerisr.c:128: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/esp32_timerisr.c:129: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [bab7a9d0858e64b3405e34d55b5099ab02833306] - Code Rule Check (C++) OK.

File: os/arch/xtensa/src/esp32/esp32_timerisr.c [hunk 2]
```diff
@@ -115,7 +115,7 @@ static inline uint32_t xtensa_getcompare(void)
```
Comment (junmin-kim) [inline os/arch/xtensa/src/esp32/esp32_timerisr.c]: Is there a reason to change file permissions?

Comment (sunghan-chang) [reply os/arch/xtensa/src/esp32/esp32_timerisr.c]: As @junmin-kim said, please revert a change of file permission.

Comment (ChinaTizenRT) [reply os/arch/xtensa/src/esp32/esp32_timerisr.c]: reverted the file permission.

File: os/arch/xtensa/src/esp32/esp32_timerisr.c [hunk 3]
```diff
@@ -125,8 +125,8 @@ int64_t get_instant_time(void)
 	compare = xtensa_getcompare();
 	diff = xtensa_getcount() - compare;
 	int ticks = clock_systimer();
-    int64_t ms = TICK2MSEC(ticks) + (diff*1000) / g_tick_divisor;
-    return ms;
+	int64_t ms = TICK2MSEC(ticks) + (diff*1000) / g_tick_divisor;
```
Comment (sunghan-chang) [inline os/arch/xtensa/src/esp32/esp32_timerisr.c]: Actually you should not include two different kinds of change in a commit.
Could you split coding rules changes from this commit and fix one more violation?
```
(diff * 1000)
```
needs spaces around `*`

Comment (ChinaTizenRT) [reply os/arch/xtensa/src/esp32/esp32_timerisr.c]: done.

**Comment by seinfra** [comment]
Target : [9abf589d225f64d32dacfc2ef29581e65319c6b8] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9abf589d225f64d32dacfc2ef29581e65319c6b8] - Code Rule Check Result:
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxO)
os/arch/xtensa/src/esp32/esp32_timerisr.c:118: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxW)
os/arch/xtensa/src/esp32/esp32_timerisr.c:128: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/xtensa/src/esp32/esp32_timerisr.c:129: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [51858da5abee3435695da7a891447c30f03be88a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [51858da5abee3435695da7a891447c30f03be88a] - Code Rule Check OK.

**Pull request merged**